mirdata/rwc_classical.py

```diff
@@ -0,0 +1,233 @@
+        """.format(save_path))
+
+    # metadata
+    download_path = utils.download_from_remote(
+            METADATA_REMOTE, data_home=annotations_path, force_overwrite=force_overwrite)
+    utils.unzip(download_path, metadata_path, cleanup=True)
+
+
+def exists(data_home=None):
+    save_path = utils.get_save_path(data_home)
+    dataset_path = os.path.join(save_path, DATASET_DIR)
+    return os.path.exists(dataset_path)
+
+
+def validate(dataset_path, data_home=None):
+    missing_files, invalid_checksums = utils.validator(
+        INDEX, data_home, dataset_path)
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    return list(INDEX.keys())
+
+
+def load(data_home=None):
+    save_path = utils.get_save_path(data_home)
+    dataset_path = os.path.join(save_path, DATASET_DIR)
+    validate(dataset_path, data_home)
+    rwc_classical_data = {}
+    for key in track_ids():
+        rwc_classical_data[key] = Track(key, data_home=data_home)
+    return rwc_classical_data
+
+
+def _load_sections(sections_path):
+    if not os.path.exists(sections_path):
+        return None
+    begs = []  # timestamps of section beginnings
+    ends = []  # timestamps of section endings
+    secs = []  # section labels
+
+    with open(sections_path, 'r') as fhandle:
+            reader = csv.reader(fhandle, delimiter='\t')
+            for line in reader:
+                begs.append(float(line[0])/100.0)
+                ends.append(float(line[1])/100.0)
+                secs.append(line[2])
+
+    return utils.SectionData(np.array(begs), np.array(ends), np.array(secs))
+
+
+def _load_beats(beats_path):
+    if not os.path.exists(beats_path):
+        return None
+    beat_times = []   # timestamps of beat interval beginnings
+    beat_positions = []  # beat position inside the bar
+    mapping_positions = {'48': '1', '96': '2', '144': '3', '384': '4'}
+
+    with open(beats_path, 'r') as fhandle:
+        reader = csv.reader(fhandle, delimiter='\t')
+        for line in reader:
+            beat_times.append(float(line[0])/100.0)
+            beat_positions.append(mapping_positions[line[2]])
+
+    return utils.BeatData(np.array(beat_times), np.array(beat_positions))
+
+
+def _load_metadata(data_home):
+
+    metadata_path = utils.get_local_path(
+        data_home, os.path.join(
+            DATASET_DIR, 'metadata-master', 'rwc-c.csv'
+        )
+    )
+
+    if not os.path.exists(metadata_path):
+        raise OSError('Could not find {}'.format(metadata_path))
+
+    with open(metadata_path, 'r', encoding='utf-8') as fhandle:
+        dialect = csv.Sniffer().sniff(fhandle.read(1024))
+        fhandle.seek(0)
+        reader = csv.reader(fhandle, dialect)
+        raw_data = []
+        for line in reader:
+            if line[0] != 'Piece No.':
+                raw_data.append(line)
+
+    metadata_index = {}
+    for line in raw_data:
+        if line[0] == 'Piece No.':
+            continue
+        p = '00' + line[0].split('.')[1][1:]
```

Comment:
move this and the below logic to the for loop in line 181, and remove the variable `raw_data`